src/Microsoft.ML.Tokenizers/Model/SentencePieceBaseModel.cs

```diff
@@ -0,0 +1,751 @@
+            }
+
+            static bool AppendTokenWithCheckingPrefix(bool addDummyPrefix, bool treatWhitespaceAsSuffix, string token, char prefixSuffixChar, Span<char> destination, ref bool prefixRemoved, ref int suffixIndex, ref int idsConsumed, ref int charsConsumed)
+            {
+                if (token.Length == 0)
+                {
+                    return true;
+                }
+
+                Span<char> buffer = destination.Slice(charsConsumed);
+
+                ReadOnlySpan<char> tokenSpan = token.AsSpan();
+
+                if (!addDummyPrefix)
+                {
+                    if (tokenSpan.Length > buffer.Length)
+                    {
+                        return false;
+                    }
+
+                    if (prefixSuffixChar != ' ')
+                    {
+                        for (int i = 0; i < tokenSpan.Length; i++)
+                        {
+                            buffer[i] = tokenSpan[i] == prefixSuffixChar ? ' ' : tokenSpan[i];
+                        }
+                    }
+                    else
+                    {
+                        tokenSpan.CopyTo(buffer);
+                    }
+
+                    buffer = buffer.Slice(tokenSpan.Length);
+                    charsConsumed += tokenSpan.Length;
+                    idsConsumed++;
+                    return true;
+                }
+
+                if (treatWhitespaceAsSuffix)
+                {
+                    if (tokenSpan[tokenSpan.Length - 1] == prefixSuffixChar)
+                    {
+                        suffixIndex = charsConsumed + tokenSpan.Length - 1;
+                    }
+
+                    if (tokenSpan.Length > buffer.Length)
+                    {
+                        return false;
+                    }
+
+                    if (prefixSuffixChar != ' ')
+                    {
+                        for (int i = 0; i < tokenSpan.Length; i++)
+                        {
+                            buffer[i] = tokenSpan[i] == prefixSuffixChar ? ' ' : tokenSpan[i];
+                        }
+                    }
+                    else
+                    {
+                        tokenSpan.CopyTo(buffer);
+                    }
+
+                    charsConsumed += tokenSpan.Length;
+
+                    idsConsumed++;
+                }
+                else
+                {
+                    int delta = !prefixRemoved && token[0] == prefixSuffixChar ? 1 : 0;
+                    if (buffer.Length < token.Length - delta)
+                    {
+                        return false;
+                    }
+
+                    tokenSpan = tokenSpan.Slice(delta);
+                    if (prefixSuffixChar != ' ')
+                    {
+                        for (int i = 0; i < tokenSpan.Length; i++)
+                        {
+                            buffer[i] = tokenSpan[i] == prefixSuffixChar ? ' ' : tokenSpan[i];
+                        }
```

Comment:
Ditto. You could add a polyfill extension Replace method somewhere for downlevel use, and then just use Replace at these call sites.